pkg/cmd/roachtest/test_runner.go

```diff
@@ -892,6 +897,11 @@ func (r *testRunner) runWorker(
 				c = nil
 			}
 		}
+
+		failureRate := float64(failedTestCount) / float64(totalTestCount)
+		if failureRate > roachtestflags.AutokillThreshold {
+			return errors.Errorf("failure rate %.2f exceeds limit %.2f", failureRate, roachtestflags.AutokillThreshold)
```

Comment:
Same as above, this seems like we are failing workers, but we want to fail the entire run.

Reply:
Added a `testCount` in the `testRunner` struct which keeps track of the total number of tests in the entire run and used that to calculate the `failureRate`